### PR TITLE
fix: `im.counters` returns `null` for unread msgs for user who never opened the DM

### LINKED ISSUE #33301: Unexpected behaviour in im.counters endpont API
### Description:

When querying the im.counters endpoint for a user that has never entered the chat, the unreads counter returns null instead of the total number of messages in the room.

### Steps to reproduce:

1. Open a direct message conversation between two users
2. Send some messages from one of the users
3. Query the im.counters endpoint with the userId of the other user of the conversation

### Expected behavior:

The unreads variable shows the same number of messages as the msgs variable.

### Actual behavior:

The unreads variable returns null

### Server Setup Information:

- Version of Rocket.Chat Server: 6.10.2
- License Type: Community
- Number of Users: 479
- Operating System: Debian 12 (bookworm)
- Deployment Method: docker
- Number of Running Instances: 1

### PR DESCRIPTION
**Pull request by KevLehman**
## Issue(s)

https://rocketchat.atlassian.net/browse/CORE-832
## Steps to test or reproduce

- Crete user A & B
- Send message with user A to User B
- Without opening DM with user B, request `im.counters`
- It will return `null` for unread counter even when user has never opened the DM
## Further comments

Fixes #33301

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/direct-message.ts [hunk 1]
```diff
@@ -343,6 +343,112 @@ describe('[Direct Messages]', () => {
...
+					.query({
+						roomId: testDM._id,
+					})
+					.expect('Content-Type', 'application/json')
+					.expect(200)
+					.expect((res) => {
+						console.log(res.body);
+						expect(res.body).to.have.property('success', true);
+						expect(res.body).to.have.property('joined', true);
+						expect(res.body).to.have.property('members').and.to.be.an('array').and.to.be.eq(2);
```
Comment (KevLehman) [inline apps/meteor/tests/end-to-end/api/direct-message.ts:441]:
```suggestion
						expect(res.body).to.have.property('members').and.to.be.a('number').and.to.be.eq(2);
```

File: apps/meteor/tests/end-to-end/api/direct-message.ts [hunk 2]
```diff
@@ -343,6 +343,112 @@ describe('[Direct Messages]', () => {
...
+				await request
+					.get(api('im.counters'))
+					.set(userCreds)
+					.query({
+						roomId: testDM._id,
+					})
+					.expect('Content-Type', 'application/json')
+					.expect(200)
+					.expect((res) => {
+						console.log(res.body);
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/end-to-end/api/direct-message.ts:438]:
```suggestion
```

File: apps/meteor/tests/end-to-end/api/direct-message.ts [hunk 3]
```diff
@@ -343,6 +343,112 @@ describe('[Direct Messages]', () => {
...
+						expect(res.body).to.have.property('unreadsFrom');
+						expect(res.body).to.have.property('msgs').and.to.be.a('number').and.to.be.eq(1);
+						expect(res.body).to.have.property('latest');
+						expect(res.body).to.have.property('userMentions').and.to.be.a('number').and.to.be.eq(0);
+					});
+			});
+		});
+	});
+
 	it('/im.counters', (done) => {
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/end-to-end/api/direct-message.ts:452]: Can we move this one to the new `describe`?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: apps/meteor/tests/end-to-end/api/direct-message.ts (hunk 3 above)
Comment (KevLehman) [reply apps/meteor/tests/end-to-end/api/direct-message.ts]: its actually there with another name
i just forgot to remove it 😬

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**